src/views/SystemProfile/index.tsx

```diff
@@ -614,6 +668,14 @@ const SystemProfile = () => {
       </div>
     </MainContent>
   );
+
+  return modalProps ? (
+    <Modal title={t('System Profile')} isOpen={isOpen} closeModal={closeModal}>
```

Comment:
These props have ts errors due to `isOpen` and `closeModal` possibly undefined.  You can maybe do something like this to have default values:

```
const { id, isOpen = false, closeModal = () => null } = modalProps || {};
```